### PR TITLE
Event consolidation service implementation

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#1639](https://github.com/ballerina-platform/ballerina-standard-library/issues/1639)

## Checklist
- [X] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: websubhub-examples/kafka-hub/consolidator/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+target
```
Comment (ldclakmal) [inline websubhub-examples/kafka-hub/consolidator/.gitignore:1]: Shall we move the .gitignore file into the project root (`websubhub-example` directory)?

File: websubhub-examples/kafka-hub/consolidator/consolidator_service.bal [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import ballerina/log;
+import consolidatorService.config;
+import consolidatorService.util;
+import consolidatorService.connections as conn;
+import consolidatorService.persistence as persist;
+
+listener kafka:Listener kafkaListener = check new (config:KAFKA_BOOTSTRAP_NODE, conn:subscribersConsumerConfig);
+
+service kafka:Service on kafkaListener {
+    remote function onConsumerRecord(kafka:Caller caller, kafka:ConsumerRecord[] records) returns error? {
```
Comment (shafreenAnfar) [inline websubhub-examples/kafka-hub/consolidator/consolidator_service.bal:29]: Does `onConsumerRecord` event get triggered in parallel or sequentially ?

File: websubhub-examples/kafka-hub/consolidator/init_consolidator.bal [hunk 3]
```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (shafreenAnfar) [inline websubhub-examples/kafka-hub/consolidator/init_consolidator.bal:1]: Let's have this in a separate branch

File: websubhub-examples/kafka-hub/consolidator/modules/config/configurations.bal [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+// under the License.
+
+import consolidatorService.util;
+
+# IP and Port of the Kafka bootstrap node
+public configurable string KAFKA_BOOTSTRAP_NODE = "localhost:9092";
+
+# Kafka topic which will get notified for websub topic registration/deregistration
+# All the hubs must be pointed to the same Kafka topic to notify websub topic registration/deregistration
+public configurable string REGISTERED_TOPICS_TOPIC = "registered-topics";
```
Comment (shafreenAnfar) [inline websubhub-examples/kafka-hub/consolidator/modules/config/configurations.bal:24]: Is this comment correct ?

File: websubhub-examples/kafka-hub/consolidator/modules/config/configurations.bal [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+# Kafka topic which stores consolidated websub topics for the hub
+public configurable string CONSOLIDATED_TOPICS_TOPIC = "consolidated-topics";
+
+# Kafka topic which will get notified for websub subscription/unsubscription
+# All the hubs must be pointed to the same Kafka topic to notify websub subscription/unsubscription
+public configurable string SUBSCRIBERS_TOPIC = "registered-subscribers";
```
Comment (shafreenAnfar) [inline websubhub-examples/kafka-hub/consolidator/modules/config/configurations.bal:31]: Is this comment correct ?

File: websubhub-examples/kafka-hub/consolidator/modules/config/configurations.bal (hunk 4 above)
Comment (ayeshLK) [reply websubhub-examples/kafka-hub/consolidator/modules/config/configurations.bal:24]: Yes, this is the `topic` which every hub notifies for topic registration/deregistration.

File: websubhub-examples/kafka-hub/consolidator/modules/config/configurations.bal (hunk 5 above)
Comment (ayeshLK) [reply websubhub-examples/kafka-hub/consolidator/modules/config/configurations.bal:31]: Yes, this is the `topic` which every hub notifies for subscription/unsubscription

File: websubhub-examples/kafka-hub/consolidator/modules/config/configurations.bal [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+# Kafka topic which is stores consolidated websub subscribers for this server
+public configurable string CONSOLIDATED_SUBSCRIBERS_TOPIC = "consolidated-subscribers";
```
Comment (shafreenAnfar) [inline websubhub-examples/kafka-hub/consolidator/modules/config/configurations.bal:34]: consolidated-websub-subscribers

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**